se_mailbox/smaildir.py

```diff
@@ -36,19 +43,17 @@ def iter_folders(self):
     def remove_folder(self, folder):
         """Delete the named folder, which must be empty."""
         path = os.path.join(self._path, '.' + folder)
-
-        def raise_if_not_empty(subpath):
-            """Raise a NotEmptyError if the folder is not empty."""
-            for entry in scandir.scandir(os.path.join(path, subpath)):
-                if len(entry.name) < 1 or entry.name[0] != '.':
-                    raise mailbox.NotEmptyError(
-                        'Folder contains message(s): %s' % folder)
-
-        raise_if_not_empty("new")
-        raise_if_not_empty("cur")
+        for entry in scandir.scandir(os.path.join(path, 'new')):
```

Comment:
This was changed in order to make unit testing easy?

Reply:
why the need to duplicate code?

Reply:
@RoxanaButuc unanswered query here.

Reply:
This was change in order to get the package up to date with mail-core (smaildir from se-mailboxes and the one from mail-core).